source/GameInterface/Services/ItemRosters/ItemRosterRegistry.cs

```diff
@@ -0,0 +1,23 @@
+﻿using GameInterface.Services.Registry;
+using System.Threading;
+using TaleWorlds.CampaignSystem.Roster;
+
+namespace GameInterface.Services.ItemRosters
+{
+    internal class ItemRosterRegistry : RegistryBase<ItemRoster>
+    {
+        private const string ItemRosterIdPrefix = "CoopItemRoster";
+        private static int ItemRosterCounter = 0;
+        public ItemRosterRegistry(IRegistryCollection collection) : base(collection) { }
+
+        public override void RegisterAll()
+        {
+            //TODO
```

Comment:
Please register all ItemRosters. You can use the CampaignObjectManager to get all objects with itemrosters